src/Agent.Worker/Release/ReleaseFileSystemManager.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release
+{
+    [ServiceLocator(Default = typeof(ReleaseFileSystemManager))]
+    public interface IReleaseFileSystemManager : IAgentService
```

Comment:
Some things to consider:
1) GetFiles has exactly one caller and is a two-liner (can be simplified by using DirectoryInfo.EnumerateFiles).
2) FileExists has one caller and is a one-liner.
3) WriteStreamToFile: a) you're not opening the file in async mode so you're not actually doing asynchronous IO. There is an overload ctor that accepts an `isAsync` bool. b) the Stream class has a CopyToAsync method, you don't need to do the while loop and buffer management yourself.
4) GetFileReader: FileNotFoundException might be more appropriate than ArgumentOutOfRangeException
5) _retryExecutor  is never used.